Alias `test` to `provides` and `stub` to `consumes` for Specmatic Config.

StarKhan6368 opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

**What**: Alias `test` to `exposes` and `stub` to `depends`

<!-- Why are these changes necessary? -->

**Why**: 

<!-- How were these changes implemented? -->

**How**:  Using the `@JsonAlias` annotation.

<!-- Have you done all of these things?  -->

**Checklist**:

<!-- add "N/A" to the end of each line that's irrelevant to your changes -->

<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->

- [x] Documentation added to the README.md OR link to PR on https://github.com/specmatic/specmatic-documentation
- https://github.com/znsio/specmatic-documentation/pull/66
- [x] Tests
- [x] Tested with Sample Projects
- [ ] Sonar Quality Gate

<!-- feel free to add additional comments -->

StarKhan6368 committed:
Alias 'test' to 'exposes' and 'stub' to 'depends' in Specmatic config.

StarKhan6368 committed:
Add test configs for `exposes` and `depends` aliases.

joelrosario commented on core/src/main/kotlin/in/specmatic/core/SpecmaticConfig.kt:
```diff
@@ -76,7 +73,9 @@ data class Source(
     val branch: String? = null,
+    @field:JsonAlias("exposes")
     val test: List<String>? = null,
+    @field:JsonAlias("depends")
```
This should be `dependencies`

joelrosario reviewed (changes requested):
@StarKhan6368 can you test these in the sample projects as well by updating specmatic.yml / json to use dependencies and exposes (both specmatic-order-bff-java and specmatic-order-api-java)

StarKhan6368 committed:
Rename `depends` alias to `dependencies`.

StarKhan6368 committed:
Fix tests with renamed `dependencies` alias.

joelrosario commented on core/src/test/resources/specmaticConfigFiles/specmatic_exp_dep.json:
```diff
@@ -0,0 +1,83 @@
+      "provider": "git",
+      "repository": "https://contracts",
+      "exposes": [
+        "com/petstore/1.spec"
```
Can you use yaml instead of spec for the extensions, as .spec is the legacy format.

StarKhan6368 replied:
Renamed OAS file extensions to .yaml instead of legacy .spec, updated the test expectations as well.

joelrosario reviewed (changes requested):
I've added a comment requesting changes.

joelrosario committed:
Merge branch 'main' into alias-test-stub

StarKhan6368 committed:
Rename OAS files to .yaml, replacing legacy .spec extension.

joelrosario committed:
Merge remote-tracking branch 'origin/alias-test-stub' into alias-test-stub

StarKhan6368 committed:
Rename aliases to `provides` and `consumes`.

- Revert older OAS test to use `.spec` extension.
- Implement a new test for aliases.
- Rename the resources used in alias tests.

joelrosario committed:
Merge remote-tracking branch 'origin/alias-test-stub' into alias-test-stub

joelrosario committed:
Updated the version to 1.3.39

joelrosario reviewed (approved):
LGTM
